lib/ttnt/tasks.rb

```diff
@@ -0,0 +1,54 @@
+require 'rake/testtask'
+require 'rake/file_list'
+require 'ttnt/testtask'
+require 'ttnt/test_selector'
+require 'rugged'
+
+# Reference: https://github.com/bundler/bundler/blob/master/lib/bundler/gem_helper.rb
+module TTNT
+  class TaskHelper
+    include Rake::DSL if defined? Rake::DSL
+
+    def self.install_tasks
+      new.install
+    end
+
+    def install
+      namespace :ttnt do
+        desc 'Generate test-to-code mapping for current commit object'
+        task 'anchor' do
+          # TODO: what if multiple test tasks are defined?
+          test_task = TTNT::TestTask.instances.first
+          test_files = []
+          test_files += test_task.test_files.to_a if test_task.test_files
+          test_files += Rake::FileList[test_task.pattern] if test_task.pattern
+
+          # TODO: properly regard run options defined for Rake::TestTask
+          gem_root = File.expand_path('..', File.dirname(File.expand_path(__FILE__)))
+          args = "-I#{gem_root} -r ttnt/anchor"
+          args += " -I#{test_task.libs.join(':')}" unless test_task.libs.empty?
```

Comment:
I don't know any interesting resources about the load path that would be up to date but you can easily see that yes, the load path [is reset](https://gist.github.com/robin850/6b4e3a64920561ab3703) when you run a ruby process from Ruby (Rake's `ruby` method relies on the `Kernel#system` one to spawn a ruby process).

Also you don't have to add the `-I#{gem_root}` because RubyGems monkey-patches `require` to work with gems so you don't have to add them to the load path.